### PR TITLE
Play ansible roles on host

### PR DESCRIPTION
**Pull request by lhellebr**
Add a method to play ansible roles on a host.
Example:
```
host = api.Host().search(query={'search': f'name={rex_contenthost.hostname}'})[0]
host.play_roles()
```

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -4332,6 +4332,25 @@ def bulk_available_incremental_updates(self, synchronous=True, timeout=None, **k
         response = client.post(self.path('bulk/available_incremental_updates'), **kwargs)
         return _handle_response(response, self._server_config, synchronous, timeout)
 
+    def play_roles(self, synchronous=True, timeout=None, **kwargs):
```
Comment (Gauravtalreja1) [inline nailgun/entities.py:4335]: Would suggest to rename this func be specific, and mind moving this func to the end of the class where all ansible_role related func exists for readability?
```suggestion
    def play_ansible_roles(self, synchronous=True, timeout=None, **kwargs):
```

File: nailgun/entities.py [hunk 2]
```diff
@@ -4332,6 +4332,25 @@ def bulk_available_incremental_updates(self, synchronous=True, timeout=None, **k
...
+        """Play all assigned ansible roles
```
Comment (Gauravtalreja1) [inline nailgun/entities.py:4336]: As same endpoint for both host and host group, so it'd be nice to cover this func for hostgroup too, wdyt?
```suggestion
        """Play all assigned ansible roles on host
```

Comment (lhellebr) [reply nailgun/entities.py]: It would. I did what I needed for my tests, there's a lot of things that would be nice.

File: nailgun/entities.py [hunk 3]
```diff
@@ -4883,6 +4884,25 @@ def remove_ansible_role(self, synchronous=True, timeout=None, **kwargs):
...
+            Defaults to ``nailgun.entity_mixins.TASK_TIMEOUT``.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all content decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.post(self.path('play_roles'), **kwargs)
```
Comment (Gauravtalreja1) [inline nailgun/entities.py:4903]: Can you include a unittest for this func/path?

Comment (Gauravtalreja1) [reply nailgun/entities.py:4903]: Thanks, Can you add play_ansible_roles to PathTestCase::test_id_and_which?

Comment (lhellebr) [reply nailgun/entities.py:4903]: Done

**Review by Gauravtalreja1 — APPROVED** [review]

File: nailgun/entities.py [hunk 4]
```diff
@@ -4883,6 +4884,25 @@ def remove_ansible_role(self, synchronous=True, timeout=None, **kwargs):
...
+    def play_ansible_roles(self, synchronous=True, timeout=None, **kwargs):
+        """Play all assigned ansible roles on a Host
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param timeout: Maximum number of seconds to wait until timing out.
+            Defaults to ``nailgun.entity_mixins.TASK_TIMEOUT``.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all content decoded.
```
[the remark this answers is not in the document]
Comment (lhellebr) [reply nailgun/entities.py:4896]: That's probably copypasta, I'll fix it

**Review by sambible — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**